packages/cli/src/commands/deploy/index.ts

```diff
@@ -509,8 +523,8 @@ export default async (client: Client): Promise<number> => {
 
     const createArgs: CreateOptions = {
       name,
-      env: deploymentEnv,
-      build: { env: deploymentBuildEnv },
+      env: deploymentEnv as Dictionary<string>,
+      build: { env: deploymentBuildEnv as Dictionary<string> },
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏Do we need the type assertions? Can we get rid of them for stronger type safety?

Reply:
I tried playing around with it some more but because environment variables _can_ be undefined we'd end up doing something like
```
// If there's any undefined values, then inherit them from this process
  try {
    await addProcessEnv(log, deploymentEnv);
    await addProcessEnv(log, deploymentBuildEnv);
  } catch (err: unknown) {
    error(errorToString(err));
    return 1;
  }

  // Filter out any remaining undefined values
  const filteredDeploymentEnv: Dictionary<string> = Object.fromEntries(
    Object.entries(deploymentEnv).filter(
      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- _ is supposed to be unused
      ([_, value]) => value !== undefined
    )
  ) as Dictionary<string>;
  const filteredDeploymentBuildEnv: Dictionary<string> = Object.fromEntries(
    Object.entries(deploymentBuildEnv).filter(
      // eslint-disable-next-line @typescript-eslint/no-unused-vars -- _ is supposed to be unused
      ([_, value]) => value !== undefined
    )
  ) as Dictionary<string>;
```
which kinda ends up doing the same thing unless my TS-fu is missing something